src/test/java/com/puppycrawl/tools/checkstyle/DefaultLoggerTest.java

```diff
@@ -278,6 +276,50 @@ public void testNullArgs() throws Exception {
                 "Violation should contain exception info, but was " + outputForNullArgs);
     }
 
+    @Test
+    public void testNewCtor() throws Exception {
+        final ResourceBundle bundle = ResourceBundle.getBundle(
+                Definitions.CHECKSTYLE_BUNDLE, Locale.ROOT);
+        final String auditStartedMessage = bundle.getString(DefaultLogger.AUDIT_STARTED_MESSAGE);
+        final String auditFinishedMessage = bundle.getString(DefaultLogger.AUDIT_FINISHED_MESSAGE);
+        final String addExceptionMessage = new MessageFormat(bundle.getString(
+                DefaultLogger.ADD_EXCEPTION_MESSAGE), Locale.ROOT).format(new String[] {"myfile"});
+        final String infoOutput;
+        final String errorOutput;
+        try (MockByteArrayOutputStream infoStream = new MockByteArrayOutputStream()) {
+            try (MockByteArrayOutputStream errorStream = new MockByteArrayOutputStream()) {
+                final DefaultLogger dl = new DefaultLogger(
+                        infoStream, OutputStreamOptions.CLOSE,
+                        errorStream, OutputStreamOptions.CLOSE);
+                dl.auditStarted(null);
+                dl.addException(new AuditEvent(5000, "myfile"),
+                        new IllegalStateException("upsss"));
+                dl.auditFinished(new AuditEvent(6000, "myfile"));
+                infoOutput = infoStream.toString(StandardCharsets.UTF_8.name());
+                errorOutput = errorStream.toString(StandardCharsets.UTF_8.name());
+
```

Comment:
Assert that the streams are closed already.
This should be done in the `try` block.